torch/testing/_internal/common_device_type.py

```diff
@@ -311,11 +315,11 @@ def instantiated_test(self, name=name, test=test_fn, dtype=dtype, op=op):
                 instantiate_test_helper(cls, name, test=test, dtype=dtype, op=None)
 
 
-class CPUTestBase(DeviceTypeTestBase):
+class CPUTestBase(Generic[DeviceTypeTestBase]):
```

Comment:
this is not how you should approach this. the exceptions shown in type hint indicates that the functions & attributes are not properly typed. You should approach this by adding syntax sugar (a.k.a. [PEP-484](https://www.python.org/dev/peps/pep-0484/)) to the methods/variable/attributes.

For example. `DeviceTypeTestBase` should have `device_type: str = ...` instead of `device_type = ...`